From dd1f7ab8a88ddc8fbb304aaca25f549e26fa8529 Mon Sep 17 00:00:00 2001
From: Markus Elfring <elfring@users.sourceforge.net>
Date: Wed, 23 Aug 2017 09:45:06 +0200
Subject: [PATCH] ALSA: timer: Adjust a condition check in snd_timer_resolution()
Git-commit: dd1f7ab8a88ddc8fbb304aaca25f549e26fa8529
Patch-mainline: v4.14-rc1
References: bsc#1121278

The script "checkpatch.pl" pointed information out like the following.

Error: do not use assignment in if condition

Thus fix the affected source code place.

Signed-off-by: Markus Elfring <elfring@users.sourceforge.net>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 sound/core/timer.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/core/timer.c b/sound/core/timer.c
index a9b9a277e00c..6051aed3197f 100644
--- a/sound/core/timer.c
+++ b/sound/core/timer.c
@@ -393,7 +393,8 @@ unsigned long snd_timer_resolution(struct snd_timer_instance *timeri)
 
 	if (timeri == NULL)
 		return 0;
-	if ((timer = timeri->timer) != NULL) {
+	timer = timeri->timer;
+	if (timer) {
 		if (timer->hw.c_resolution)
 			return timer->hw.c_resolution(timer);
 		return timer->hw.resolution;
-- 
2.20.1

